File: 2024 November/Second Largest/main.cpp
// Link: https://www.geeksforgeeks.org/problems/second-largest3735/1

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int getSecondLargest(vector<int> &arr)
    {
        int maxi = *max_element(arr.begin(), arr.end());
        int ans = -1;
        for (auto it : arr)
        {
            if (it < maxi)
            {
                ans = max(ans, it);
            }
        }
        return ans;
    }
};

int main()
{
    Solution object;
    vector<int> arr = {12, 35, 1, 10, 34, 1};
    cout << object.getSecondLargest(arr) << endl;

    arr = {10, 5, 10};
    cout << object.getSecondLargest(arr) << endl;

    arr = {10, 10, 10};
    cout << object.getSecondLargest(arr) << endl;
    return 0;
}

File: 2024 November/Second Largest/main.py
# Link: https://www.geeksforgeeks.org/problems/second-largest3735/1


class Solution:
    def getSecondLargest(self, arr):
        maxi = max(arr)
        ans = -1
        for it in arr:
            if it < maxi:
                ans = max(ans, it)
        return ans


if __name__ == "__main__":
    object = Solution()
    arr = [12, 35, 1, 10, 34, 1]
    print(object.getSecondLargest(arr))

    arr = [10, 5, 10]
    print(object.getSecondLargest(arr))

    arr = [10, 10, 10]
    print(object.getSecondLargest(arr))
